rules.py

```diff
@@ -688,12 +693,41 @@ def check(self, element):
         if element_label:
             if element_label in self.labels:
                 raise base.ElectionError(
-                    "Line %d. Duplicate label '%s'. Label already defined earlier"
-                     % (element.sourceline, element_label))
+                    "Line %d. Duplicate label '%s'. Label already defined"
+                    % (element.sourceline, element_label))
             else:
                 self.labels.add(element_label)
 
 
+class ReusedCandidate(base.TreeRule):
+    """Candidate should be referred to by only one contest.
+
+    A Candidate object should only ever be referenced from one contest. If a
+    Person is running in multiple Contests, then that Person is a Candidate
+    several times over, but a Candida(te|cy) can't span contests.
+    """
+    seen_candidates = {} # mapping of candidates and candidate selections
+
+    def check(self):
+        candidate_selections = self.get_elements_by_class(
```

Comment:
I did that since we also don't want candidateselections within the same contest to refer to the same candidate. This way the user will get info on which candidateselections specifically have an issue.